src/common/kube-helpers.ts

```diff
@@ -71,67 +131,44 @@ export function loadFromOptions(options: KubeConfigOptions): KubeConfig {
   return kc;
 }
 
-export function loadConfig(pathOrContent?: string): KubeConfig {
-  return loadConfigFromString(
-    fse.pathExistsSync(pathOrContent)
-      ? readResolvedPathSync(pathOrContent)
-      : pathOrContent
-  );
+export interface ConfigResult {
```

Comment:
While I really like monadic return types (especially coming from my liking of rust). I don't think this place would be good. Because it really should be (if I was in rust) `(KubeConfig, Option<Joi.ValidationError>)`, which isn't that more ergonomic than what I have now.

Reply:
Makes sense.